### PR TITLE
Don't recommend reformatting merge commits

### PR DESCRIPTION
**Pull request by cbaylis**

### CODE REVIEW

File: git-pre-commit-format [hunk 1]
```diff
@@ -313,14 +313,37 @@ else
     cat "$patch"
 fi
 
+# We don't want to suggest applying clang-format after merge resolution
+if git rev-parse MERGE_HEAD >/dev/null 2>&1; then
+    THIS_IS_A_MERGE=true
```
Comment (barisione) [inline git-pre-commit-format]: Can  you please make it `readonly this_is_a_merge=true`?

File: git-pre-commit-format [hunk 2]
```diff
@@ -313,14 +313,37 @@ else
...
 echo "${b}The staged content is not formatted correctly.${n}"
 echo "The patch shown above can be applied automatically to fix the formatting."
 echo
 
+if $THIS_IS_A_MERGE; then
+    echo "${b}You appear to be committing the result of a merge. It is not${n}"
+    echo "${b}recommended to apply the patch if it will reformat any code you${n}"
+    echo "${b}did not modify in your branch.${n}"
+    echo
+    RECOMMEND_APPLY=" (not recommended for merge!)"
```
Comment (barisione) [inline git-pre-commit-format]: Same here about readonly and lower case please (for all the variables).

File: git-pre-commit-format [hunk 3]
```diff
@@ -313,14 +313,37 @@ else
...
+else
+    THIS_IS_A_MERGE=false
+fi
+
 echo
 echo "${b}The staged content is not formatted correctly.${n}"
 echo "The patch shown above can be applied automatically to fix the formatting."
 echo
 
+if $THIS_IS_A_MERGE; then
```
Comment (barisione) [inline git-pre-commit-format]: Maybe we should skip the “Will commit anyway!” message later if this is a merge.

**Comment by cbaylis** [comment]
All requested changes done.

**Pull request merged**